arxiv/config/__init__.py

```diff
@@ -0,0 +1,167 @@
+"""Flask configuration."""
+import importlib.metadata
+from typing import Optional
+import os
+import secrets
+from urllib.parse import urlparse
+
+SERVER_NAME = None
+
+SECRET_KEY = os.environ.get("SECRET_KEY", secrets.token_hex(16))
+A11Y_URL = os.environ.get(
+    "ARXIV_ACCESSIBILITY_URL", "mailto:web-accessibility@cornell.edu"
+)
+
+EXTERNAL_URL_SCHEME = os.environ.get("EXTERNAL_URL_SCHEME", "https")
+BASE_SERVER = os.environ.get("BASE_SERVER", "arxiv.org")
+
+AUTH_SERVER = os.environ.get("AUTH_SERVER", BASE_SERVER)
+"""Hostname for auth paths like /login and /logout.
+
+Usually the same as BASE_SERVER but can be configured.
+"""
+
```

Comment:
for completeness RSS also has a server (although I doubt anyone outside of RSS cares)